tungsten_ci_utils/nexus_cleanup/clean_up.groovy

```diff
@@ -55,11 +55,16 @@ if (components != null) {
         def tag = comp.version();
         def tagSplited = tag.split("-");
         def build_number = tagSplited[tagSplited.length - 1];
-        def retentionList = tagList.subList(0, tagList.size() - retentionCount);
-
+        def retentionList = tagList;
+        if(tagList.size() > retentionCount){
+            retentionList = tagList.subList(0, tagList.size() - retentionCount);
+        } else {
+            log.info("Component skipped: ${comp.name()} ${comp.version()}");
```

Comment:
Change the info to something more meaningful which would say why was the skip done. E.g. 'Number of builds is already less than $retentionCount'